### PR TITLE
ROX-21656: roxctl deployment check E2E tests

### PR DESCRIPTION
**Pull request by Maddosaurus**
## Description
Add E2E groovy tests for the new features in `roxctl deployment check` that were introduced in [ROX-20907](https://issues.redhat.com/browse/ROX-20907).
Namely, the tests assert on the addition of correct remarks. 
Remarks consist of the following information per deployment:
- Permission level
- Applied Network Policies

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [ ] ~Evaluated and added CHANGELOG entry if required~
- [ ] ~Determined and documented upgrade steps~
- [ ] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Running groovy tests on CI and manually against OS4 cluster

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by Maddosaurus** [comment]
In the first run a different test failed, not the ones introduced here

File: qa-tests-backend/src/test/groovy/DeploymentCheck.groovy [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        serviceAccount = new K8sServiceAccount(
+                name: "check-deployment-sa",
+                namespace: DEPLOYMENT_CHECK,
+        )
+        orchestrator.createServiceAccount(serviceAccount)
+
+        def orchRoles = orchestrator.getClusterRoles()
+        for (K8sRole r : orchRoles) {
+            if (r.getName() == "cluster-admin") {
+                clusterAdmin = r
```
[the remark this answers is not in the document]
Comment (Maddosaurus) [reply qa-tests-backend/src/test/groovy/DeploymentCheck.groovy:58]: We could create our own role, but what I want to ensure is that the permission level the deployments run as is `cluster_admin`. As it is a default account, I don't expect it to change/vanish anytime soon. And if that were the case, we have an `assert` in place to fail the test if the account wasn't found or doesn't exist.
So, let's leave it for now and act on it, should we need to 😄

File: qa-tests-backend/src/test/groovy/DeploymentCheck.groovy [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        DetectionServiceOuterClass.DeployDetectionResponse res
+
+        when:
+        withRetry(20, 5) {
+            res = DetectionService.getDetectDeploytimeFromYAML(req)
+        }
+        log.info "Got remarks:\n ${res.remarksList}"
+
+        then:
+        assert res
```
[the remark this answers is not in the document]
Comment (Maddosaurus) [reply qa-tests-backend/src/test/groovy/DeploymentCheck.groovy:98]: The assignment of `res` happens in the `withRetry` block in [line 93](https://github.com/stackrox/stackrox/pull/10157/files/379929c075aba7e57d19055ac9117a6e9d268976#diff-b5f56c38463372680b435e95c99911aa6d54ecae26962ac58854e2d22e3c3c05R93), so we have a retry with 20 tries for `res` at this point

File: qa-tests-backend/src/test/groovy/DeploymentCheck.groovy [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
```
[the remark this answers is not in the document]
Comment (Maddosaurus) [reply qa-tests-backend/src/test/groovy/DeploymentCheck.groovy:126]: Same as above - we have the assignment of `res` surrounded in a `withRetry` block, here in line 121 😄

File: qa-tests-backend/src/test/groovy/DeploymentCheck.groovy [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        assert res.getRemarksList().size() == 2
+        assert !res.getRemarksList().findAll { it.getName() == DEPLOYMENT_CHECK }.isEmpty()
+        assert !res.getRemarksList().findAll { it.getName() == "de2" }.isEmpty()
+    }
+
+    static String createDeploymentYaml(String deploymentName) {
```
[the remark this answers is not in the document]
Comment (Maddosaurus) [reply qa-tests-backend/src/test/groovy/DeploymentCheck.groovy]: Very good point! Done in [98f93c5](https://github.com/stackrox/stackrox/pull/10157/commits/98f93c5336f7dc9e6e916790a9d8f87cd9749986)

**Review by lvalerom — APPROVED** [review]
LGTM 🚀

**Comment by stehessel** [comment]
@davdhacs I concur, good catch :)

**Pull request merged**